tests/test_pay.py

```diff
@@ -6065,3 +6067,513 @@ def test_pay_remember_hint(node_factory):
     # We should not have touched fw1, and should succeed after a single call
     p = sender.rpc.pay(inv)
     assert(p['parts'] == 1)
+
+
+def test_injectpaymentonion(node_factory, executor):
```

Comment:
This is a terrible test, and will likely be one of the first ones to break. While bunching up a whole bunch of scenarios may seem ideal, as we ammortize the startup, it doesn't quite work that well. For one thing the longer the test the more state we accumulate over time. With state comes state drift, so as we change the underlying implementation some of the implicit assumptions may no longer be true, and reconstructing that subtle drift and fixing it can be hard, especially if it's been some time or if you're not the original author.
Furthermore, even if such a test fails with a true negative, the signal we get from it is muddied by the fact that it may be any of the bundled scenarios, also requiring the reconstruction of the rationale, and figuring out which aspect broke.

All this to say, that shorter, less busy tests provide better signal in case there is a regression, and are less hard to keep non-flaky.

Reply:
Fair! I kept adding to it as I implemented, and ended up folding it together into one giant test. 

I'll split it, though in this case they're basically unrelated and unlikely to have cascading failure.